crates/edr_rpc_server/src/node.rs

```diff
@@ -437,4 +618,548 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn block_number() -> Result<()> {
+        let fixture = NodeTestFixture::new().await?;
+
+        let block_number = fixture.node.block_number().await;
+        assert_eq!(block_number, 0);
+
+        fixture.node.lock_data().await.mine_block(None).await?;
+        let block_number = fixture.node.block_number().await;
+        assert_eq!(block_number, 1);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn block_and_total_difficulty_by_block_spec_tags() -> Result<()> {
+        async fn assert_block_number(
+            fixture: &NodeTestFixture,
+            block_tag: BlockTag,
+            expected_block_number: u64,
+        ) {
+            let block = fixture
+                .node
+                .block_and_total_difficulty_by_block_spec(&BlockSpec::Tag(block_tag))
+                .await;
+
+            assert_eq!(block.unwrap().block.header().number, expected_block_number);
+        }
```

Comment:
Since multiple functions use this, it'd be better to avoid code duplication by moving it one layer up (inside the `mod tests`)

Reply:
They are all slightly different.